### PR TITLE
#1666 Added error message for cost and description of products on finance form

### LINKED ISSUE #1666: Finance Fixes - Add Error Messaging to Cost and Description of Products on Finance Form
### Description

Currently the description and cost fields on the create/edit reimbursement request forms get highlighted red if there's a form error, but they don't actually display what the error is. These fields should display error messages right beneath them exactly how all the other fields in the form do.

### Acceptance Criteria

- [ ] Description and cost fields on a reimbursement product in a reimbursement request create/edit form shows the error message if beneath it if there's an error

### PR DESCRIPTION
**Pull request by jessicayzhao**
## Changes

Displays error messages below cost and description fields for products in finance form.

## Screenshots

Screenshot 2023-10-25 at 7 20 03 PM

Screenshot 2023-10-25 at 7 20 36 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1666

### CODE REVIEW

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx [hunk 1]
```diff
@@ -97,41 +98,51 @@ const ReimbursementProductTable: React.FC<ReimbursementProductTableProps> = ({
...
+                                    startAdornment: <InputAdornment position="start">$</InputAdornment>
+                                  }}
+                                  onBlur={(e) => onCostBlurHandler(parseFloat(e.target.value), product.index)}
+                                  sx={{ width: '100%' }}
+                                  error={!!errors.reimbursementProducts?.[product.index]?.cost}
+                                />
+                              )}
+                            />
+                            <FormHelperText error>
+                              {errors.reimbursementProducts?.[product.index]?.cost?.message}
```
Comment (callmejim1226) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx:143]: I realized we get the following long non-user-friendly error msg under the cost box when we delete the whole value and try to rewrite the cost value. Can we fix this, so the cost box error msg stays the same as `Amount must be greater than 0`?

image

**Review by callmejim1226 — CHANGES_REQUESTED** [review]
Great work!!

I left a few comments about the cost box error message and div usage.

**Review by callmejim1226 — APPROVED** [review]
LGTM! Great job 💯

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**